cmd/log/src/lib.rs

```diff
@@ -0,0 +1,338 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! ## `humility log`
+//!
+//! `humility log` reads and displays any Hubris string buffers (as created
+//! via the `stringbuf!` macro in the Hubris `ringbuf` crate).  e.g.:
+//!
+//! ```console
+//! % humility log -m LOG_RINGBUF
+//! humility: attached via J-Link
+//! humility: ring buffer task_jefe::LOG_RINGBUF in jefe:
+//! instruction
+//! Task #2 Memory fault at address 0x0
+//! Task #2 Illegal instruction
+//! Task #2 Memory fault at address 0x0
+//! Task #2 Illegal instruction
+//! Task #2 Memory fault at address 0x0
+//! Task #2 Illegal instruction
+//! ...
+//! ```
+//!
+//! Use `-m` or `--monitor` to continuously monitor the buffer, otherwise will just print the
+//! current log and exit.
+//!
+//! If an argument is provided, only string buffers that have a name that
+//! contains the argument as a substring, or are in a task that contains
+//! the argument as a substring will be displayed.  
+//!
+//! See the [`ringbuf`
+//! documentation](https://github.com/oxidecomputer/hubris/blob/master/lib/ringbuf/src/lib.rs) for more details.
+
+use anyhow::{bail, Result};
+use clap::Command as ClapCommand;
+use clap::{CommandFactory, Parser};
+use humility::core::Core;
+use humility::hubris::*;
+use humility::reflect::{self, Load, Value};
+use humility_cmd::doppel::{StaticCell, Stringbuf};
+use humility_cmd::{Archive, Attach, Command, Run, Validate};
+use std::collections::BTreeMap;
+use std::iter::zip;
+use std::num::ParseIntError;
+use std::str;
+use std::thread;
+use std::time;
+
+#[derive(Parser, Debug)]
+#[clap(name = "stringbuf", about = env!("CARGO_PKG_DESCRIPTION"))]
+struct StringbufArgs {
+    /// list variables
```

Comment:
Does the `variables` mean all `Stringbuf`s that are defined?

Reply:
Yes, I updated the doc to reflect this.